test/validation/mod2c_core/cpp/NaSm.cpp

```diff
@@ -357,10 +357,7 @@ static inline void initmodel(_threadargsproto_) {
   m = m0;
  {
    rates ( _threadargscomma_ v ) ;
-   #pragma omp critical
-   {
-    m = minf + 1.1 ;
-   }
+   m = minf + 1.1 ;
```

Comment:
@iomaganaris : have to think a bit just a quick comment: here I was thinking of OpenMP threads active from coreneuron side as well.

Reply:
I thought those were separate than the mechanisms.. I need to remind myself how that works

Reply:
You were right!

I went back and looked into generated code:

```cpp
#define pv  _nt->_data[_ppvar[0*_STRIDE]]
#define pia _nt->_data[_ppvar[1*_STRIDE]]
#define pim _nt->_data[_ppvar[2*_STRIDE]]
....

 if ( ri > 0.0 ) {
     #pragma omp atomic update
     #pragma acc atomic update
     pim = pim - ia ;
 }
```

So global variable like `pim` are inside `_nt->_data`. So even if there will be multiple OpenMP threads active from CoreNEURON side, they would be operating on separate `NrnThread`. (I was thinking of `pim` being "C global" variable).

So your change make sense to me!